lib/interactive_viewer.ml

```diff
@@ -2,85 +2,98 @@ open Nottui
 module W = Nottui_widgets
 open Lwd_infix
 
-let index_v = Lwd.var 0
-let index = Lwd.get index_v
-
-let current_patch patches =
-  let$ index = index in
-  List.nth_opt patches index
+let operation_info z_patches : ui Lwd.t =
+  let$ z = Lwd.get z_patches in
+  W.string
+    ~attr:Notty.A.(fg lightcyan)
+    (Printf.sprintf "Operation %d of %d"
+       (Zipper.get_current_index z + 1)
+       (Zipper.get_total_length z))
 
 let ui_of_operation operation =
   let green_bold_attr = Notty.A.(fg green ++ st bold) in
   let red_bold_attr = Notty.A.(fg red ++ st bold) in
   let blue_bold_attr = Notty.A.(fg blue ++ st bold) in
   match operation with
   | Patch.Create path ->
-    Ui.hcat [
-          W.string "Creation of ";
-          W.string ~attr:green_bold_attr path; 
-    ]
+      Ui.hcat [ W.string "Creation of "; W.string ~attr:green_bold_attr path ]
   | Patch.Delete path ->
-    Ui.hcat [
-      W.string "Deletion of ";
-      W.string ~attr:red_bold_attr path;
-    ]
+      Ui.hcat [ W.string "Deletion of "; W.string ~attr:red_bold_attr path ]
   | Patch.Rename (old_path, new_path) ->
-    Ui.hcat [
-      W.string "Rename with modifications ";
-      W.string ~attr:blue_bold_attr old_path;
-      W.string " to ";
-      W.string ~attr:green_bold_attr new_path;
-    ]
+      Ui.hcat
+        [
+          W.string "Rename with modifications ";
+          W.string ~attr:blue_bold_attr old_path;
+          W.string " to ";
+          W.string ~attr:green_bold_attr new_path;
+        ]
   | Patch.Rename_only (old_path, new_path) ->
-    Ui.hcat [
-      W.string "Rename ";
-      W.string ~attr:blue_bold_attr old_path;
-      W.string " to ";
-      W.string ~attr:green_bold_attr new_path;
-    ]
+      Ui.hcat
+        [
+          W.string "Rename ";
+          W.string ~attr:blue_bold_attr old_path;
+          W.string " to ";
+          W.string ~attr:green_bold_attr new_path;
+        ]
   | Patch.Edit path ->
-    Ui.hcat [
-      W.string "Modification of ";
-      W.string ~attr:blue_bold_attr path;
-    ]
+      Ui.hcat
+        [ W.string "Modification of "; W.string ~attr:blue_bold_attr path ]
 
 let string_of_hunk = Format.asprintf "%a" Patch.pp_hunk
 
-let current_operation patches =
-  let$ current_patch = current_patch patches in
-  match current_patch with
-  | Some p -> ui_of_operation p.Patch.operation
-  | None -> W.string "No operation"
+let current_operation z_patches : ui Lwd.t =
+  let$ z = Lwd.get z_patches in
+  let p = Zipper.get_focus z in
+  ui_of_operation p.Patch.operation
+
+let current_hunks z_patches : ui Lwd.t =
+  let$ z = Lwd.get z_patches in
+  let p = Zipper.get_focus z in
+  let hunks =
+    List.map
+      (fun h -> W.string ~attr:Notty.A.(fg lightblue) (string_of_hunk h))
```

Comment:
Hmm, why did you decide to change the color of the hunks? :thinking:

Reply:
I think it was a personal preference, but I'll remove it for now until we establish an official colour!